src/Magento/FunctionalTestingFramework/DataGenerator/Persist/CurlHandler.php

```diff
@@ -201,11 +201,26 @@ private function resolveUrlReference($urlIn, $entityObjects)
 
         if (!empty($matchedParams)) {
             foreach ($matchedParams[0] as $paramKey => $paramValue) {
+                $paramEntityParent = "";
+                $matchedParent = [];
+                $dataItem = $matchedParams[1][$paramKey];
+                preg_match_all("/(.+?)\./", $dataItem, $matchedParent);
+
+                if (!empty($matchedParent) && !empty($matchedParent[0])) {
```

Comment:
Add comment:
`Assign entity "type" for filtering, remove "type." from parameter`